src/NUnitFramework/framework/Constraints/TypeConstraint.cs

```diff
@@ -61,6 +61,12 @@ protected TypeConstraint(Type type, string descriptionPrefix)
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             actualType = actual == null ? null : actual.GetType();
+
+            if (actual is Exception)
```

Comment:
Like you, I see this as a code smell. But I don't have a better suggestion and I can see us opening this up to more types in the future.